docs/Datagrid.md

```diff
@@ -0,0 +1,737 @@
+
+The `Datagrid` component renders a list of records as a table. It is usually used as a descendant of the [`<List>`](#the-list-component) and [`<ReferenceManyField>`](./Fields.md#referencemanyfield) components. Outside these components, it must be used inside a `ListContext`.
+
+## Usage
+
+`<Datagrid>` renders as many columns as it receives `<Field>` children. It uses the field `label` as column header (or, for fields with no `label`, the field `source`).
+
+```jsx
+// in src/posts.js
+import * as React from "react";
+import { List, Datagrid, TextField, EditButton } from 'react-admin';
+
+export const PostList = () => (
+    <List>
+        <Datagrid>
+            <TextField source="id" />
+            <TextField source="title" />
+            <TextField source="body" />
+            <EditButton />
+        </Datagrid>
+    </List>
+);
+```
+
+You can find more advanced examples of `<Datagrid>` usage in the [demos](./Demos.md). 
+
+
+**Tip**: To let users hide or show columns at will, check the [`<SelectColumnsButton>`](https://marmelab.com/ra-enterprise/modules/ra-preferences#selectcolumnsbutton-store-datagrid-columns-in-preferences)<img class="icon" src="./img/premium.svg" />, an [Enterprise Edition](https://marmelab.com/ra-enterprise) component.
+
+The `<Datagrid>` is an **iterator** component: it gets an array of records from the `ListContext`, and iterates to display each record in a row. Other examples of iterator component are [`<SimpleList>`](#the-simplelist-component) and [`<SingleFieldList>`](#the-singlefieldlist-component).
+
+## Props
+
+Here are all the props accepted by the component:
+
+* [`body`](#body)
+* [`bulkActionButtons`](#bulkActionButtons)
+* [`children`](#children)
+* [`empty`](#empty)
+* [`expand`](#expand)
+* [`header`](#header)
+* [`hover`](#hover)
+* [`isRowExpandable`](#isrowexpandable)
+* [`isRowSelectable`](#isrowselectable)
+* [`optimized`](#performance)
+* [`rowStyle`](#rowstyle)
+* [`rowClick`](#rowclick)
+* [`size`](#size)
+* [`sx`](#sx-css-api)
+
+Additional props are passed down to [the material-ui `<Table>` element](https://material-ui.com/api/table/).
+
+## `body`
+
+By default, `<Datagrid>` renders its body using `<DatagridBody>`, an internal react-admin component. You can pass a custom component as the `body` prop to override that default. And by the way, `<DatagridBody>` has a `row` prop set to `<DatagridRow>` by default for the same purpose. `<DatagridRow>` receives the row `record`, the `resource`, and a copy of the `<Datagrid>` children. That means you can create custom `<Datagrid>` logic without copying several components from the react-admin source.
+
+For instance, the `<Datagrid isRowSelectable>` prop allows to hide the selection checkbox for some records. To show a *disabled* checkbox instead of hiding it, you can override `<DatagridRow>` and `<DatagridBody>` as follows:
+
+```jsx
+// in src/PostList.js
+import { Datagrid, DatagridBody, List, TextField } from 'react-admin';
```

Comment:
```suggestion
import * as React from 'react';
import { Datagrid, DatagridBody, List, TextField } from 'react-admin';
```

Reply:
I've removed the import react from all snippets because:

1. it's implicit
2. it's not required in recent versions of CRA
3. It adds unnecessary noise in these snippets